### PR TITLE
docs: move dtl to peerdeps for rtl

### PR DESCRIPTION
**Pull request by MatanBobi**
Docs for https://github.com/testing-library/react-testing-library/pull/1305

Explanation that we need to manually install `@testing-library/dom` as it's moved to peer dependency.

**Do not merge before merging the RTL PR**

### CODE REVIEW

File: docs/react-testing-library/intro.mdx [hunk 1]
```diff
@@ -9,19 +9,22 @@ APIs for working with React components.
 
 ## Installation
 
+To get started with `React Testing Library`, you'll need to install it together
+with it's peerDependency `@testing-library/dom`:
```
Comment (eps1lon) [inline docs/react-testing-library/intro.mdx:13]:
```suggestion
with its peerDependency `@testing-library/dom`:
```

File: docs/react-testing-library/intro.mdx [hunk 2]
```diff
@@ -9,19 +9,22 @@ APIs for working with React components.
...
 ```
 
 ### With TypeScript
 
-To get full type coverage, you need to install the types for `react-dom` as well:
+To get full type coverage, you need to install the types for `react-dom` as
+well:
 
 ```bash npm2yarn
 npm install --save-dev @testing-library/react @types/react-dom
```
Comment (eps1lon) [inline docs/react-testing-library/intro.mdx:25]:
```suggestion
npm install --save-dev @testing-library/react @testing-library/dom @types/react-dom
```

Comment (MatanBobi) [reply docs/react-testing-library/intro.mdx:25]: Missed that, good point. Thanks :)

**Comment by MatanBobi** [comment]
@eps1lon I think we need to merge this one too :)

**Review by eps1lon — APPROVED** [review]

**Pull request merged**